### PR TITLE
add ribbon bar implementation for extenstion

### PR DESCRIPTION
**Pull request by nickanna42**
This PR makes some small, non-breaking requests to the ribbon panel implementation of ES. Specifically, it allows for the host application to provide React components to the client extensions via an attribute on the ES object that is passed to the extensions.

### CODE REVIEW

**Comment by itechify** [comment]
Is this PR updating an example extension to show how this can currently be done? Just wanted to confirm that is the case since I don't see any changes to es-runtime / es-home.

**Comment by nickanna42** [comment]
> Is this PR updating an example extension to show how this can currently be done? Just wanted to confirm that is the case since I don't see any changes to es-runtime / es-home.

This is all the code that is needed. The frontend framework implements the spec defined in the api beforing passing the extension-scaffold object to the activate function and places the host elements on the page according to the static application resources for the app.

I have updated it to allow for the implementation to optionally add react components onto the extension-scaffold object for the convience of the extensions that implement the `activate` spec to have access to consistently themed components provided by the frontend framework.

File: es-runtime/src/es-api.ts [hunk 1]
```diff
@@ -169,7 +170,7 @@ export interface RibbonBar {
     showRibbonTab: (id: string) => HTMLDivElement | null
     hideRibbonTab: (id: string) => HTMLDivElement | null
     components?: {
-        [key: string]: HTMLElement
```
Comment (moesol) [inline es-runtime/src/es-api.ts]: We have users of ES that use Angular or other non-React frameworks.

We do not want to be tied to React in the API. We do want to allow React components to use the API and to be ribbon items, etc. We have examples of that use case.

**Review by moesol — CHANGES_REQUESTED** [review]

File: es-runtime/package.json [hunk 2]
```diff
@@ -21,7 +21,9 @@
     "build"
   ],
   "dependencies": {
-    "events": "^3.3.0"
+    "events": "^3.3.0",
+    "react": "^17.0.2",
```
Comment (moesol) [inline es-runtime/package.json:25]: We do not want to add react as a dependency.

Nothing in es-runtime requires it.
If extensions want to use React, they can import it.

**Review by moesol — CHANGES_REQUESTED** [review]

**Comment by nickanna42** [comment]
I was able to overload the typing in applications/extensions in lieu of this PR.

**Pull request closed without merging**